modules/oe_content_featured_media_field/tests/src/Kernel/FeaturedMediaFieldTest.php

```diff
@@ -0,0 +1,240 @@
+    $this->installEntitySchema('media');
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('node');
+    $this->installEntitySchema('file');
+    $this->installConfig([
+      'media',
+      'image',
+      'file',
+      'system',
+      'oe_media',
+      'oe_content_featured_media_field',
+    ]);
+
+    // Create an account with permission and set it as current user.
+    $account = $this->createUser(['view media']);
+    $this->container->get('current_user')->setAccount($account);
+
+    // Create an image file.
+    $this->container->get('file_system')->copy($this->root . '/core/misc/druplicon.png', 'public://example.jpg');
+    $image = File::create(['uri' => 'public://example.jpg']);
+    $image->save();
+
+    // Create a media entity of image media type.
+    $this->mediaEntity = Media::create([
+      'bundle' => 'image',
+      'name' => 'Test image',
+      'field_media_image' => [
+        [
+          'target_id' => $image->id(),
+          'alt' => 'default alt',
+          'title' => 'default title',
+        ],
+      ],
+    ]);
+    $this->mediaEntity->save();
+
+    // Create a content type.
+    $type = NodeType::create(['name' => 'Test content type', 'type' => 'test_ct']);
+    $type->save();
+
+    $this->fieldStorage = FieldStorageConfig::create([
+      'field_name' => 'featured_media_field',
+      'entity_type' => 'node',
+      'type' => 'oe_featured_media',
+      'cardinality' => 1,
+      'entity_types' => ['node'],
+    ]);
+    $this->fieldStorage->save();
+
+    $this->field = FieldConfig::create([
+      'label' => 'Featured media field',
+      'field_name' => 'featured_media_field',
+      'entity_type' => 'node',
+      'bundle' => 'test_ct',
+      'settings' => [
+        'handler' => 'default:media',
+        'handler_settings' => [
+          'target_bundles' => [
+            'image' => 'image',
+          ],
+        ],
+        'sort' => [
+          'field' => '_none',
+        ],
+        'auto_create' => '0',
+      ],
+      'required' => FALSE,
+    ]);
+    $this->field->save();
+  }
+
+  /**
+   * Test the featured media field.
+   */
+  public function testFeaturedMediaField(): void {
+    $values = [
+      'type' => 'test_ct',
+      'title' => 'My node title',
+      'featured_media_field' => [
+        [
+          'target_id' => $this->mediaEntity->id(),
+          'caption' => 'Image caption text',
+        ],
+      ],
+    ];
+
+    // Create a node.
+    $node = Node::create($values);
+    $node->save();
+
+    $entity_type_manager = $this->container->get('entity_type.manager')->getStorage('node');
+    $entity_type_manager->resetCache();
+    /** @var \Drupal\node\NodeInterface $node */
+    $node = $entity_type_manager->load($node->id());
+
+    $expected = [
+      [
+        'target_id' => $this->mediaEntity->id(),
+        'caption' => 'Image caption text',
+      ],
+    ];
+    // Assert the base field values.
```

Comment:
Let's remove the word base, it's misleading indeed. Assert the field values should be enough. Also why we test the node label? Let's drop it.